File: ch200203/settlement-api/src/main/java/com/hanwha/settlement/settlements/controller/SettlementController.java
package com.hanwha.settlement.settlements.controller;

import com.hanwha.settlement.settlements.dto.CreateRequest;
import com.hanwha.settlement.settlements.dto.SettlementReceivesResponse;
import com.hanwha.settlement.settlements.dto.TransferRequest;
import com.hanwha.settlement.settlements.model.Settlement;
import com.hanwha.settlement.settlements.service.SettlementService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/settlements")
@RequiredArgsConstructor
public class SettlementController {

    private final SettlementService settlementService;

    @PostMapping
    public ResponseEntity<Settlement> requestSettlement(@RequestBody CreateRequest request) {
        Settlement settlement = settlementService.request(request);
        return ResponseEntity.ok(settlement);
    }

    /**
     * 전달 받은 금액을 정산한다.
     */
    @PostMapping("/{settlementReceiveId}/transfer")
    public ResponseEntity<Void> transferPay(@PathVariable Long settlementReceiveId, @RequestBody TransferRequest request) {
        settlementService.transferPay(settlementReceiveId, request);
        return ResponseEntity.ok().build();
    }

    @GetMapping("/settlementReceives/{userId}")
    public ResponseEntity<List<SettlementReceivesResponse>> getSettlementReceives(@PathVariable Long userId) {
        List<SettlementReceivesResponse> settlementReceivesResponses = settlementService.getSettlementReceives(userId);
        return ResponseEntity.ok().body(settlementReceivesResponses);
    }

}

File: mjkim103301/pay-separately/src/main/kotlin/org/example/pay/controller/SettlementController.kt
package org.example.pay.controller

import org.example.pay.dto.SettlementDto
import org.example.pay.dto.request.SettlementCreateDto
import org.example.pay.dto.request.SettlementUpdateDto
import org.example.pay.service.CalculateSettlementService
import org.example.pay.service.RemitSettlementService
import org.example.pay.service.RequestSettlementSerivce
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/settlements")
class SettlementController @Autowired constructor(
    private val remitSettlementService: RemitSettlementService,
    private val requestSettlementService: RequestSettlementSerivce,
    private val calculateSettlementService: CalculateSettlementService
) {


    /**
     * 보험료 정산 요청
     */
    @PostMapping("/request")
    fun createSettlement(@RequestBody settlementCreateDto: SettlementCreateDto): ResponseEntity<SettlementDto> {
        val settlementDto = requestSettlementService.createRequestedSettlements(settlementCreateDto)
        return ResponseEntity.ok(settlementDto)
    }

    /**
     * 정산금 송금
     */
    @PostMapping("/remit")
    fun remitSettlement(@RequestBody settlementUpdateDto: SettlementUpdateDto): ResponseEntity<Unit> {
        remitSettlementService.remitSettlement(settlementUpdateDto.settlementDetailId)
        return ResponseEntity.ok().build()
    }


    /**
     * 정산 요청한 항목 납입 완료 확인
     */
    @PutMapping("/{id}")
    fun checkSettlementIsCompleted(@PathVariable id: Long): ResponseEntity<Boolean> {
        val isCompleted = requestSettlementService.checkSettlementIsCompleted(id)
        return ResponseEntity.ok(isCompleted)
    }

}
